test/e2e/test.js

```
@@ -35,118 +34,169 @@ var SCOPES = [
+          assert.ok(breakpoint.id, 'breakpoint should have an id');
+          assert.ok(breakpoint.location, 'breakpoint should have a location');
+          assert.strictEqual(breakpoint.location.path, 'test.js');
+          return { debuggee: debuggee, breakpoint: breakpoint };
+        });
+        return result;
+      })
+      .then(function(result) {
+        assert.ok(result.breakpoint);
+        assert.ok(result.debuggee);
+        console.log('-- waiting a bit before running fib');
+        return Q.delay(20 * 1000).then(function() { return result; });
+      })
+      .then(function(result) {
+        console.log('-- Running fib');
+        fib(12);
+        console.log('-- waiting before checking if the breakpoint was hit');
+        return Q.delay(10 * 1000).then(function() { return result; });
+      })
+      .then(function(result) {
+        console.log('-- now checking if the breakpoint was hit');
+        var promise = getBreakpoint(result.debuggee, result.breakpoint);
+        return promise;
+      })
+      .then(function(body) {
+        console.log('-- results of get breakpoint\n', body);
+        assert.ok(body.breakpoint, 'should have a breakpoint in the response');
+        var hit = body.breakpoint;
+        assert.ok(hit.isFinalState, 'breakpoint should have been hit');
+        assert.ok(Array.isArray(hit.stackFrames), 'should have stack ');
+        var top = hit.stackFrames[0];
+        assert.ok(top, 'should have a top entry');
+        assert.ok(top.function, 'frame should have a function property');
+        assert.strictEqual(top.function, 'fib');
+
+        var arg = _.find(top.arguments, function(a) {
+          return a.name === 'n';
+        });
+        assert.ok(arg, 'should find the n argument');
+        assert.strictEqual(arg.value, '10');
+        console.log('Test passed');
+        return true;
+      })
+      .catch(function(err) {
+        console.error(err);
+        process.exit(2);
```

I can let it throw out of the promise. I was originally contemplating doing something with the error, but this doesn't seem necessary anymore.